.github/workflows/build.yml

```diff
@@ -0,0 +1,57 @@
+name: Build
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  build:
+    container: ghcr.io/zeldaret/ss-build:main
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        version: [SOUE01]
+
+    steps:
+    # Checkout the repository (shallow clone)
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        submodules: recursive
+
+    # Set Git config
+    - name: Git config
+      run: git config --global --add safe.directory "$GITHUB_WORKSPACE"
+
+    # Copy the original files to the workspace
+    - name: Prepare
+      run: cp -R /orig .
+
+    # Build the project
+    - name: Build
+      run: |
+        python configure.py --map --version ${{ matrix.version }} \
+            --binutils /binutils --compilers /compilers
+        ninja all_source build/${{ matrix.version }}/progress.json
+
+    # Upload progress if we're on the main branch
+    - name: Upload progress
+      if: github.ref == 'refs/heads/main'
+      continue-on-error: true
+      env:
+        PROGRESS_SLUG: ss
+        PROGRESS_API_KEY: ${{ secrets.PROGRESS_API_KEY }}
+      run: |
+        python tools/upload_progress.py -b https://progress.decomp.club/ \
+          -p $PROGRESS_SLUG -v ${{ matrix.version }} \
+          build/${{ matrix.version }}/progress.json
+
+    # Upload map files
+    - name: Upload map
+      uses: actions/upload-artifact@v3
+      with:
+        name: ${{ matrix.version }}_maps
+        path: build/${{ matrix.version }}/**/*.MAP
```

Comment:
What's this for?

Reply:
Each build will push the generated linker map files as artifacts that can be downloaded from CI. It's useful in case someone wants to grab the latest map for import into Ghidra, Dolphin or similar.